packages/proto-signing/src/codec/cosmos/bank/v1beta1/tx.ts

```diff
@@ -0,0 +1,324 @@
+  fromAddress: string;
+  toAddress: string;
+  amount: Coin[];
+}
+
+/**
+ *  MsgSendResponse defines the Msg/Send response type.
+ */
+export interface MsgSendResponse {}
+
+/**
+ *  MsgMultiSend represents an arbitrary multi-in, multi-out send message.
+ */
+export interface MsgMultiSend {
+  inputs: Input[];
+  outputs: Output[];
+}
+
+/**
+ *  MsgMultiSendResponse defines the Msg/MultiSend response type.
+ */
+export interface MsgMultiSendResponse {}
+
+const baseMsgSend: object = {
+  fromAddress: "",
+  toAddress: "",
+};
+
+const baseMsgSendResponse: object = {};
+
+const baseMsgMultiSend: object = {};
+
+const baseMsgMultiSendResponse: object = {};
+
+/**
+ *  Msg defines the bank Msg service.
+ */
+export interface Msg {
+  /**
+   *  Send defines a method for sending coins from one account to another account.
+   */
+  Send(request: MsgSend): Promise<MsgSendResponse>;
+
+  /**
+   *  MultiSend defines a method for sending coins from some accounts to other accounts.
+   */
+  MultiSend(request: MsgMultiSend): Promise<MsgMultiSendResponse>;
+}
+
+export class MsgClientImpl implements Msg {
+  private readonly rpc: Rpc;
+
+  constructor(rpc: Rpc) {
+    this.rpc = rpc;
+  }
+
+  Send(request: MsgSend): Promise<MsgSendResponse> {
+    const data = MsgSend.encode(request).finish();
+    const promise = this.rpc.request("cosmos.bank.v1beta1.Msg", "Send", data);
+    return promise.then((data) => MsgSendResponse.decode(new Reader(data)));
+  }
+
+  MultiSend(request: MsgMultiSend): Promise<MsgMultiSendResponse> {
+    const data = MsgMultiSend.encode(request).finish();
+    const promise = this.rpc.request("cosmos.bank.v1beta1.Msg", "MultiSend", data);
+    return promise.then((data) => MsgMultiSendResponse.decode(new Reader(data)));
+  }
+}
+
+interface Rpc {
+  request(service: string, method: string, data: Uint8Array): Promise<Uint8Array>;
+}
+
+export const protobufPackage = "cosmos.bank.v1beta1";
+
+export const MsgSend = {
+  encode(message: MsgSend, writer: Writer = Writer.create()): Writer {
+    writer.uint32(10).string(message.fromAddress);
+    writer.uint32(18).string(message.toAddress);
+    for (const v of message.amount) {
+      Coin.encode(v!, writer.uint32(26).fork()).ldelim();
+    }
+    return writer;
+  },
+  decode(input: Uint8Array | Reader, length?: number): MsgSend {
+    const reader = input instanceof Uint8Array ? new Reader(input) : input;
+    let end = length === undefined ? reader.len : reader.pos + length;
+    const message = { ...baseMsgSend } as MsgSend;
+    message.amount = [];
+    while (reader.pos < end) {
+      const tag = reader.uint32();
+      switch (tag >>> 3) {
+        case 1:
+          message.fromAddress = reader.string();
+          break;
+        case 2:
+          message.toAddress = reader.string();
+          break;
+        case 3:
+          message.amount.push(Coin.decode(reader, reader.uint32()));
+          break;
+        default:
+          reader.skipType(tag & 7);
```

Comment:
Okay, here if actually uses the field info (`tag & 7`) to determine how to skip the unknown. Odd this is not enforced above. eg. 

```ts
case 2:
   if tag & 7 != 2 {
       throw new Error(`Unexpected type ${tag & 7} for field 1`);
    }
    message.fromAddress = reader.string();
    break;
```

This is due to `proto-ts`, but seems like it could use improvement there
```

Reply:
Otherwise, the auto-gen code does seem more compact and less confusing. But maybe just cuz it misses some options/cases (like deciding whether to error on unknown fields or silently skip them)